src/VideoNode.h

```diff
@@ -103,18 +98,30 @@ public slots:
     // which ones that changed through the function chainsEdited.
     QList<QSharedPointer<Chain>> chains();
     void setChains(QList<QSharedPointer<Chain>> chains);
-    QSharedPointer<OpenGLWorkerContext> workerContext() const;
+
+    // A VideoNode may request of the model
+    // that certain chains exist.
+    // By default, no chains are requested.
+    // You only need to implement this method if you are
+    // writing an output. 
+    // You must also emit the signals
+    // requestedChainAdded and requestedChainRemoved.
+    // The Model will add these through setChains.
+    virtual QList<QSharedPointer<Chain>> requestedChains();
+
 protected slots:
-    virtual void chainsEdited(QList<QSharedPointer<Chain>> added, QList<QSharedPointer<Chain>> removed) = 0;
+    // If your node does anything at all, you will need to override this method
+    // and take appropriate action when the set of render chains changes.
+    virtual void chainsEdited(QList<QSharedPointer<Chain>> added, QList<QSharedPointer<Chain>> removed);
 
 protected:
-    int m_inputCount;
+    int m_inputCount{};
```

Comment:
What does this do/mean?

Reply:
it initializes it to zero (parentheses would have also been OK)
see:
https://stackoverflow.com/a/18222927